src/Shell/Task/ModelTask.php

```diff
@@ -273,10 +273,17 @@ public function findBelongsTo($model, array $associations)
                         $tmpModelName = Inflector::camelize($found);
                     }
                 }
+                $foreignKeys = (array) TableRegistry::get($tmpModelName)->primaryKey();
+                foreach ($foreignKeys as $k => $v) {
+                    if ($v === 'id') {
```

Comment:
I think for bake we can make some assumptions around the conventions that people should be following.